auth-providers/oauth2-provider/src/main/java/tech/ydb/auth/OAuth2TokenSource.java

```diff
@@ -0,0 +1,199 @@
+    public static OAuth2TokenSource fromFile(File tokenFile) {
+        return fromFile(tokenFile, JWT_TOKEN);
+    }
+
+    public static OAuth2TokenSource fromFile(File tokenFile, String tokenType) {
+        return new OAuth2TokenSource(tokenType, 24 * 60 * 60) { // Expire in 1 day
+            @Override
+            public String getToken() {
+                try (FileReader reader = new FileReader(tokenFile)) {
+                    return CharStreams.toString(reader).trim();
+                } catch (IOException e) {
+                    throw new RuntimeException("Unable to read token from " + tokenFile, e);
+                }
+            }
+        };
+    }
+
+    public static JWTTokenBuilder fromPrivateKeyPemFile(File privateKeyPemFile) {
+        try {
+            StringBuilder base64 = new StringBuilder();
+            try (BufferedReader br = new BufferedReader(new FileReader(privateKeyPemFile))) {
+                for (String line = br.readLine(); line != null; line = br.readLine()) {
+                    if (line.startsWith("-----")) {
+                        continue;
+                    }
+                    base64.append(line);
+                }
+            }
+            byte[] bytes = Base64.getDecoder().decode(base64.toString());
+            KeyFactory kf = KeyFactory.getInstance("RSA");
+            PrivateKey key = kf.generatePrivate(new PKCS8EncodedKeySpec(bytes));
+            return new JWTTokenBuilder(key);
+        } catch (IOException | NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new RuntimeException("Unable to read key from " + privateKeyPemFile, e);
+        }
+    }
+
+    public static JWTTokenBuilder fromKey(Key key) {
+        return new JWTTokenBuilder(key);
+    }
+
+    public static class JWTTokenBuilder {
+        private final Key signingKey;
+        private final SignatureAlgorithm alg;
+
+        private Clock clock = Clock.systemUTC();
+        private int ttlSeconds = 60 * 60; // 1 hour by default
+
+        private String issuer = null;
+        private String subject = null;
+        private String audience = null;
+        private String id = null;
+
+        private Map<String, Object> claims = new HashMap<>();
+
+        private JWTTokenBuilder(Key key) {
+            this.signingKey = key;
+            this.alg = SignatureAlgorithm.forSigningKey(key);
+        }
+
+        @VisibleForTesting
+        JWTTokenBuilder withClock(Clock clock) {
+            this.clock = clock;
+            return this;
+        }
+
+        public JWTTokenBuilder withId(String id) {
+            this.id = id;
+            return this;
+        }
+
+        public JWTTokenBuilder withIssuer(String issuer) {
+            this.issuer = issuer;
+            return this;
+        }
+
+        public JWTTokenBuilder withSubject(String subject) {
+            this.subject = subject;
+            return this;
+        }
+
+        public JWTTokenBuilder withAudience(String audience) {
```

Comment:
Audience is a list of strings, but for convenience it is also good to support audience as a single string (as in C++/Go/Python SDK)